### PR TITLE
bump ASO to v2.9.0

### PR DESCRIPTION
**Pull request by nojnhuh**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

This bumps the version of ASO that CAPZ compiles against and installs to [v2.9.0](https://github.com/Azure/azure-service-operator/releases/tag/v2.9.0).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [X] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
CAPZ now installs Azure Service Operator v2.9.0.
```

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -26,7 +26,6 @@ import (
 	// +kubebuilder:scaffold:imports
 	asocontainerservicev1api20210501 "github.com/Azure/azure-service-operator/v2/api/containerservice/v1api20210501"
 	asocontainerservicev1api20230201 "github.com/Azure/azure-service-operator/v2/api/containerservice/v1api20230201"
-	asocontainerservicev1api20230202preview "github.com/Azure/azure-service-operator/v2/api/containerservice/v1api20230202preview"
```
Comment (nojnhuh) [inline main.go:29]: This version was removed in the new ASO version: https://github.com/Azure/azure-service-operator/releases/tag/v2.9.0

File: config/aso/crds.yaml
Comment (nawazkh) [inline config/aso/crds.yaml:1]: > 3,929 additions, 13,536 deletions

Nice.

**Review by willie-yao — APPROVED** [review]
/lgtm

**Comment by nojnhuh** [comment]
/approve
🤠

**Pull request merged**